tests/queries/0_stateless/02763_last_day_of_week.sql

```diff
@@ -0,0 +1,17 @@
+WITH 
+    toDate('2018-12-25') + number AS d, 
+    toDate32(d) AS d32, 
+    toDateTime(d) AS dt, 
+    toDateTime64(d, 0) AS dt64
+SELECT
```

Comment:
I guess a few tests around the "edges" of the data type range would also be interesting (e.g. something similar to 02096_date_time_1970_saturation.sql but at the upper end)